add predicate functions to core and API

rschmukler opened the pull request:
Adds predicate functions for things like spec/malli validators and
run-time reflection.

henryw374 commented:
local-date? and local-time? don't work?

fyi the tick convention will be for these to be called `date?` and `time?`

can we have api tests for all preds? you can  find/replace over the ones in cljc.java-time for ease

rschmukler commented:
Oops, sorry! I named them `date?` and `time?` in core, but called them by their incorrect names in API. I'll add some tests!

rschmukler commented:
Some notes as I implemented tests:

1) Removed `offset-time?` since tick doesn't seem to have the notion of that type either

2) Renamed `day-month?` to `day-of-month?` to match the constructor.

3) Renamed `zone-id?` to `zone?` to match the constructor

4) As mentioned above, fixed calls for `date?` and `time?` in API

Sadly I can't seem to actually run the tests locally, as I get the following error on trying to require `tick.alpha.api` or `tick.core`:

```
Syntax error (FileNotFoundException) compiling at (cljc/java_time/local_date.cljc:1:1).
Could not locate cljs/java_time/interop__init.class, cljs/java_time/interop.clj or cljs/java_time/interop.cljc on classpath. Please check that namespaces with dashes use underscores in the Clojure file name.
```

So it's probably worth confirming that these tests do run / pass. Apologies if I missed something - flying a bit blind without the REPL.

henryw374 commented:
Hey, thanks that looks good. Sorry about running the tests, that should be fixed now if you pull tick master. 

Please pull that change in and run `make test`.

rschmukler commented:
Hey! Thanks for fixing master. Confirmed that I am now able to run the tests. Sorry I didn't have time to debug it myself, I saw the exclusion but figured it was intentional.

I removed `day-of-month?` as it was calling out to `month-day?` and that actually used java's `MonthDay` which tick doesn't seem to have a construct for. Beyond that it was just a bunch of little fixes.

Thanks again for the help with the PR!

henryw374 commented:
cheers
